src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2026,7 +2027,67 @@ protected override void HandleEvaluateRequestAsync(IRequestResponder<EvaluateArg
                 Type = variable.Type,
                 VariablesReference = variable.VariablesReference
             });
+        }
+
+
+        protected override void HandleReadMemoryRequestAsync(IRequestResponder<ReadMemoryArguments, ReadMemoryResponse> responder)
+        {
+            int hr;
+            ReadMemoryArguments rma = responder.Arguments;
+            ErrorBuilder eb = new ErrorBuilder(() => AD7Resources.Error_Scenario_ReadMemory);
+            try
+            {
+                if (string.IsNullOrEmpty(rma.MemoryReference))
+                {
+                    throw new ArgumentNullException("ReadMemoryArguments.MemoryReference is null or empty.");
```

Comment:
>"ReadMemoryArguments.MemoryReference is null or empty." [](start = 52, length = 55)

The argument to `ArgumentNullExecption`'s contructor is a parameter name. So you probably want `ArgumentException` instead.